processor/metricstransformprocessor/factory.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configerror"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "metrics_transform"
+)
+
+// Factory is the factory for metrics transform processor.
+type Factory struct{}
+
+// Type gets the type of the Option config created by this factory.
+func (f *Factory) Type() configmodels.Type {
+	return typeStr
+}
+
+// CreateDefaultConfig creates the default configuration for processor.
+func (f *Factory) CreateDefaultConfig() configmodels.Processor {
+	return &Config{
+		ProcessorSettings: configmodels.ProcessorSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr,
+		},
+	}
+}
+
+// CreateTraceProcessor creates a trace processor based on this config.
```

Comment:
Should you explain that this processor does not operate on traces here?

Reply:
From my understanding, this is handled by this method returning a ErrDataTypeIsNotSupported error. Since this method returns this error, the user will not be able to write this in the traces pipeline. Some of other processors also do the same for metrics. E.g. the [attribute processor](https://github.com/open-telemetry/opentelemetry-collector/blob/ba18ff826f61149404e392f58989907e71a66a44/processor/attributesprocessor/factory.go#L85)